### PR TITLE
ECS - systems with movement and jumping

### PR DESCRIPTION
**Pull request by Peter-JanGootzen**
Moet samen getest worden met https://github.com/Brick-Studios/BrickEngine/pull/14

### CODE REVIEW

File: include/controllers/game_controller.hpp [hunk 1]
```diff
@@ -2,25 +2,39 @@
...
-    void createFpsCounter(int fps);
-public:
-    GameController();
-    void gameLoop();
+    inline static const int SCREEN_HEIGTH = 720;
+    inline static const int SCREEN_WIDTH = 1280;
 };
 
-#endif
+#endif /* FILE_GAME_CONTROLLER_HPP */
```
Comment (Bram-Boris) [inline include/controllers/game_controller.hpp]: Use //

Comment (Bram-Boris) [inline include/controllers/game_controller.hpp]: //

File: include/entities/entity_factory.hpp [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    int createPanda(double x, double y) const;
+    int createGorilla(double x, double y) const;
+    int createImage(std::string path, int x, int y, int width, int height);
+private:
+    std::shared_ptr<EntityManager> entityManager;
+    RenderableFactory& renderableFactory;
+    inline static const std::string graphicsPath = "./assets/graphics/";
+};
+
+#endif /* FILE_ENTITY_FACTORY_HPP */
```
Comment (Bram-Boris) [inline include/entities/entity_factory.hpp]: Use //

File: include/systems/beast_system.hpp [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+class BeastSystem : public System {
+public:
+    BeastSystem(std::shared_ptr<EntityFactory> ef, std::shared_ptr<EntityManager> _entityManager);
+    ~BeastSystem() = default;
+private:
+    std::shared_ptr<EntityFactory> entityFactory;
+};
+
+#endif /* FILE_BEAST_SYSTEM */
```
Comment (Bram-Boris) [inline include/systems/beast_system.hpp]: Use //

File: include/systems/movement_system.hpp [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    MovementSystem(std::shared_ptr<EntityManager> entityManager, std::shared_ptr<EntityFactory> ef);
+    ~MovementSystem() = default;
+    void update(double deltatime);
+private:
+    static constexpr double TERMINAL_VELOCITY = 50'000;
+    static constexpr double JUMP_FORCE = 1'000'000;
+    static constexpr double MOVEMENT_FORCE = 5;
+};
+
+#endif /* FILE_MOVEMENT_SYSTEM_HPP */
```
Comment (Bram-Boris) [inline include/systems/movement_system.hpp]: Use //

File: src/controllers/game_controller.cpp [hunk 5]
```diff
@@ -11,66 +11,93 @@ using namespace std::chrono_literals;
...
+#include "brickengine/input/input.hpp"
+#include "brickengine/systems/rendering_system.hpp"
+#include "brickengine/systems/physics_system.hpp"
+#include "entities/layers.hpp"
+#include "systems/movement_system.hpp"
 
-GameController::GameController(){
+GameController::GameController() {
     this->delta_time = 1;
-    this->fps_cap = 120;
```
Comment (Bram-Boris) [inline src/controllers/game_controller.cpp]: Merge with master

File: include/entities/entity_factory.hpp (hunk 2 above)
Comment (LuukSantegoeds) [inline include/entities/entity_factory.hpp]: Inconsistent @JanSchollaert

File: include/systems/beast_system.hpp (hunk 3 above)
Comment (LuukSantegoeds) [inline include/systems/beast_system.hpp]: Inconsistent @JanSchollaert

File: include/systems/movement_system.hpp (hunk 4 above)
Comment (LuukSantegoeds) [inline include/systems/movement_system.hpp]: Inconsistent @JanSchollaert

**Review by LuukSantegoeds — APPROVED** [review]

File: src/systems/movement_system.cpp [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        if (!physics) continue;
+
+        //Input::getInstance().getPlayerInput(player.playerId).checkInput(BeastArenaInput::Right)
+
+        double vx = physics->vx;
+        double vy = physics->vy;
+        double mass = physics->mass;
+
+        // Moving left or right
+        if (Input::getInstance().checkInput(PlayerInput::PLAYER1_LEFT)) {
```
Comment (Bram-Boris) [inline src/systems/movement_system.cpp]: You only have to getInstance() once :)

**Review by LuukSantegoeds — APPROVED** [review]

**Review by MarkMeer — APPROVED** [review]

**Pull request merged**